itest/rust/src/virtual_methods_test.rs

```diff
@@ -463,3 +506,42 @@ fn test_notifications() {
     );
     obj.free();
 }
+
+// Used in `test_collision_object_2d_input_event` in `SpecialTests.gd`.
+#[derive(GodotClass)]
+#[class(base = RigidBody2D)]
+pub struct CollisionObject2DTest {
+    input_event_called: bool,
+    viewport: Option<Gd<Viewport>>,
+}
+
+#[godot_api]
+impl RigidBody2DVirtual for CollisionObject2DTest {
+    fn init(_: Base<RigidBody2D>) -> Self {
+        Self {
+            input_event_called: false,
+            viewport: None,
+        }
+    }
```

Comment:
This could be compiler-generted via `#[class(init, base = RigidBody2D)]`.

Reply:
yeah i mostly just wanted it to be clear exactly what they were initialized to. but since i actually check the values of these in the test it's probably not needed.